test/e2e/tests/app_test.go

```diff
@@ -48,23 +48,24 @@ func TestApp_Hash(t *testing.T) {
 		info, err := client.ABCIInfo(ctx)
 		require.NoError(t, err)
 		require.NotEmpty(t, info.Response.LastBlockAppHash, "expected app to return app hash")
+		// In next-block execution, the app hash is stored in the next block
+		blockHeight := info.Response.LastBlockHeight + 1
 
-		status, err := client.Status(ctx)
-		require.NoError(t, err)
-		require.NotZero(t, status.SyncInfo.LatestBlockHeight)
+		require.Eventually(t, func () bool {
+			status, err := client.Status(ctx)
+			require.NoError(t, err)
+			require.NotZero(t, status.SyncInfo.LatestBlockHeight)
+			return status.SyncInfo.LatestBlockHeight >= blockHeight
+		}, 60*time.Second, 3*time.Second)
```

Comment:
I'd have the interval be somewhere between 100ms and 1s, other than that this is great.